src/eventstream-proxy/eventstream-proxy.base.ts

```diff
@@ -212,7 +239,23 @@ export abstract class EventStreamProxyBase extends WebSocketEventsBase {
           !this.awaitingAck.find(msg => msg.batchNumber === inflight.batchNumber))
       ) {
         this.logger.log(`In-flight batch complete (batchNumber=${inflight.batchNumber})`);
-        this.socket.ack(inflight.batchNumber);
+        this.namespaceEventStreamSocket.get(inflight.namespace)?.ack(inflight.batchNumber);
+      }
+    }
+  }
+
+  send(namespace, payload: string) {
+    const clients = this.namespaceClients.get(namespace);
+    if (clients) {
+      // Randomly select a connected client for this namespace to distribute load
+      const selected = Math.floor(Math.random() * clients.size);
+      let i = 0;
+      for (let client of clients.keys()) {
+        if (i++ == selected) {
+          this.logger.debug(`WS <= ${payload}`);
```

Comment:
I'm always a bit wary of logging full payloads, even at debug level...

Reply:
TRACE level?